contracts/extensions/FundingQueue.sol

```diff
@@ -327,39 +345,51 @@ contract FundingQueue is DSMath, PatriciaTreeProofs {
     return fundingToTransfer;
   }
 
-  // Used for mapping backing percent to the appropriate decay rate (10 second intervals)
-  // Result of evaluating ((1 - backingPercent / 2) ** (1 / (7 * 24 * 60 * 6)))
-  //  at the following points: [0, .1, .2, .3, .4, .5, .6, .7, .8, .9, 1]
-  uint256[11] decayRates = [
-    1000000000000000000,
-    999999151896947103,
-    999998257929499257,
-    999997312851998332,
-    999996310463960536,
-    999995243363289488,
-    999994102614216063,
-    999992877291965621,
-    999991553844799874,
-    999990115177810890,
-    999988539298800050
-  ];
-
   function getDecayRate(uint256 backingPercent) internal view returns (uint256) {
     assert(backingPercent <= WAD);
 
     if (backingPercent == WAD) {
-      return decayRates[10];
+      return getDecayRateFromBin(10);
     }
 
     uint256 lowerBin = backingPercent / (10 ** 17);
     uint256 lowerPct = (backingPercent - (lowerBin * 10 ** 17)) * 10;
 
     return add(
-      wmul(decayRates[lowerBin], sub(WAD, lowerPct)),
-      wmul(decayRates[lowerBin + 1], lowerPct)
+      wmul(getDecayRateFromBin(lowerBin), sub(WAD, lowerPct)),
+      wmul(getDecayRateFromBin(lowerBin + 1), lowerPct)
     );
   }
 
+  function getDecayRateFromBin(uint256 bin) internal pure returns (uint256) {
+    // Used for mapping backing percent to the appropriate decay rate (10 second intervals)
+    // Result of evaluating ((1 - backingPercent / 2) ** (1 / (7 * 24 * 60 * 6)))
+    //  at the following points: [0, .1, .2, .3, .4, .5, .6, .7, .8, .9, 1]
```

Comment:
I think this one's my fault, but to truly make this a functional refactor only, we should check `bin<=10`, which should be assertable.